A RESTApi to get vendor Sales and buyer Orders

Shumba250 opened the pull request:
#### What does this PR do?

- Create RESTapi to get all seller sales
- Create RESTapi to get buyer orders
- The endpoint documented using swagger

#### Description of Task to be completed

- a vendor gets all the sales of the products sold
- a buyer gets all the orders of the products bought
- only the owner is allowed to update a product
#### How should this be manually tested?

- clone the repo
- switch to the fix-buyer-payment branch
- type npm install to install all packages
- npm run migrate:up and npm run db:seeds
- add STRIPE_SECRET_KEY = "sk_test_51MVA8uGaYgK6bC8Q7PwPCpkkktpHpvbt86v4cshRnJaSpWG8vjUhzND4Oxyd1rTHS0OV2bTNs9U72MnJGeEAzmBg00hZptGSrA" in .env file
- run the server by typing npm run dev
- in your browser navigate to swagger doc by typing localhost: PORT/docs/
- first add a product to the cart
- you either choose to update the cart by changing the quantity of the product
- after you can check out by navigating to the checkout endpoint
- copy the URL provided and past it in the browser to process the payment
- when the payment is successful, a receipt is send to the buyer via email.
- navigate to the sales endpoint on Swagger to get all vendor sales 
-  navigate to orders endpoint on Swagger to get all buyer orders

wakaflorien committed:
Merge pull request #49 from atlp-rwanda/ft-liveChat

#56 feature: create Users Public Live Chat

wakaflorien committed:
Merge pull request #49 from atlp-rwanda/ft-liveChat

#56 feature: create Users Public Live Chat

wakaflorien commented on src/controllers/salesController.js:
```diff
@@ -0,0 +1,18 @@
+      if (getSales.length === 0) {
+        return res.status(404).json({ status: "fail", message: "No product sold" });
+      }
+      return res.status(200).json({ data: { getSales } });
```
- follow the same format of responses here.

wakaflorien commented on src/controllers/orderController.js:
```diff
@@ -0,0 +1,21 @@
+          .status(404)
+          .json({ status: "fail", message: "No product bought" });
+      }
+      return res.status(200).json({ data: { getOrders } });
```
- follow the same format of responses here too.
```javascript
.json({status: 'success', message: 'all current orders', data: getOrders})
```

sostene12 committed:
fix(buyer-payment): tesing endpoints

- implement test endpoints
- enhance authorization on endpoints

[WIP fix-buyer-payment ]

wakaflorien commented on src/controllers/orderController.js:
```diff
@@ -0,0 +1,18 @@
+      if (getOrders.length === 0) {
+        return res.status(404).json({ status: "fail", message: "No product bought" });
+      }
+      return res.status(200).json({ status: "fail", data: { getOrders } });
```
- this status is not a fail. also missing message

wakaflorien commented on src/controllers/paymentController.js:
```diff
@@ -54,29 +53,23 @@ class PaymentController {
-      
-     return res.json({ status: "success", payment_status: session.payment_status });
+
+      return res.json({ status: "success", payment_status: session.payment_status });
```
- can you add a message attribute for more explanation

Shumba250 committed:
fix(ft-buyer-payment)
Create RESTapi to get all seller sales
Create RESTapi to get buyer orders
added a middleware that checks the owner of the vendor unpdating a product
FINISHES[fix-buyer-payment]
